pkg/cli/image/append/append.go

```diff
@@ -258,8 +259,50 @@ func (o *AppendImageOptions) Run() error {
 		}
 		fromRepo = repo
 
-		srcManifest, manifestLocation, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
-		if err != nil {
+		var srcManifest distribution.Manifest
+		var manifestLocation imagemanifest.ManifestLocation
+		multipleMatch := false
+		srcManifest, manifestLocation, err = imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+		if err == nil {
+			if ml, ok := srcManifest.(*manifestlist.DeserializedManifestList); ok {
+				if len(ml.Manifests) > 1 {
+					// if multiple matches found, try once more with an exact filter match
+					// have to do this here, need to differentiate between a filter such as 'linux/*',
+					// that may have more than one match, from 'linux/arm' that should only have a single
+					// match and should not also include linux/arm64.
+					multipleMatch = true
+					filterFn := func(d *manifestlist.ManifestDescriptor, hasMultiple bool) bool {
+						return imagemanifest.PlatformSpecString(d.Platform) == o.FilterOptions.FilterByOS
+					}
+					srcManifest, manifestLocation, err = imagemanifest.FirstManifest(ctx, from.Ref, repo, filterFn)
+					ml, ok := srcManifest.(*manifestlist.DeserializedManifestList)
+					if (ok && len(ml.Manifests) == 1) || !ok {
+						multipleMatch = false
+					}
+				}
+			}
+		}
```

Comment:
other option for the above code is to add a suggestion with the ImageRetriever error that if the filter you passed is listed below in the available os/arch options for the image, then try again but with `^,$` to avoid multiple matches.

Reply:
I'm guessing this is the option that will be preferred, so I'll push that now.